File: src/app/load_page.py
import os, subprocess
from subprocess import PIPE



def load_page(url, timeout):
    pj = os.environ.get("PROJECT_DIR")
    cmd = "phantomjs ${}/js_scripts/load_page.js --url={} --timeout={}".format(pj, url, timeout)
    p = subprocess.Popen([cmd], stdout=PIPE, stderr=PIPE)


if __name__ == '__main__':
    url = "https://www.amazon.com/s?i=arts-crafts-intl-ship&bbn=4954955011&rh=n%3A4954955011%2Cn%3A12896081%2Cn%3A12896121&dc&qid=1627957907&rnid=12896081"
    script = "/Users/anastasia/PycharmProjects/SSSystem/js_scripts/load_page.js"
    myenv = os.environ.copy()
    myenv["PATH"]=myenv["PATH"] + ":{}/phantomjs-2.1.1-macosx/bin/phantomjs".format(myenv["HOME"])
    myenv["phantomjs"] = "{}/phantomjs-2.1.1-macosx/bin/phantomjs".format(myenv["HOME"])
    os.environ["PATH"] = os.getenv("PATH")+":{}/phantomjs-2.1.1-macosx/bin/phantomjs".format(os.getenv("HOME"))
    # os.system("/Users/anastasia/phantomjs-2.1.1-macosx/bin/phantomjs {}".format(script))

    p = subprocess.Popen(["/Users/anastasia/phantomjs-2.1.1-macosx/bin/phantomjs {}".format(script)],
                         shell=True, env=myenv, stdout=PIPE, stderr=PIPE)
    # stdout = p.stdout

    out = b""
    for line in iter(p.stdout.readline, b''):
        out += line.rstrip()
        p.stdout.flush()

    #p = subprocess.Popen(["phantomjs", script, "--url={}".format(url)], shell=True)
    # p.wait()
    # out = p.stdout

    with open("/Users/anastasia/PycharmProjects/SSSystem/test_new3.html", "w") as fp:
        fp.write(out.decode())

    print("ok")

File: src/load_page.js
// load_page.path.push("")
// require("./utils.js");
// require("./main.js");

//import {parse_args, load_page} from "./utils.js";

// require("../js_scripts");


function load_page(url, timeout){
    var page = require('webpage').create();
    page.open(url, function() {

    setTimeout(function() {

        page.evaluate(function() {
            window.document.body.scrollTop = document.body.scrollHeight;
        });

        // var js = page.evaluate(function() {
        //     return document;});
        //
        // console.log(js.all[0].outerHTML);
        // phantom.exit();

    },timeout);

    var js = page.evaluate(function() {
        return document;});

    console.log(js.all[0].outerHTML);
    phantom.exit();

})

}


(function (){

    var url = "https://www.amazon.com/s?i=arts-crafts-intl-ship&bbn=4954955011&rh=n%3A4954955011%2Cn%3A12896081%2Cn%3A12896121&dc&qid=1627957907&rnid=12896081";
    var timeout = 15;

    var name = "None";
    var val = "None";

    var system = require('system');
    const args = system.args;

    args.forEach(function (arg){

        var splitted = arg.split("=");

        if (splitted.length > 1){
            name = splitted[0];
            val = splitted[1];
        }
        else (val = splitted[0]);

        if(name === "--url" || name === "-url"){
            url = val;
        }

        if(name === "--timeout" || name === "-t"){
            timeout = val;
        }
    })

    console.log(url);
    console.log(timeout);

    load_page(url, timeout);

})()